### DISCUSSION TITLE
Update content of ViewField after a TextInput changes

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#4779] [Help]
Building an invoice app.

I have added a `ViewField` that contains the invoices totals in a simple html table (no need for inputs):

`ViewField::make('Totals')->view('invoice.totals')`

Inside the the blade template I just trigger the function renderTotals:
`{!! $this->renderTotals() !!}`

That function builds the html table and returns it as a string. So far so good. But now I would like it to update once the invoices contents change (quantities, prices, positions...). 

I am aware of `afterStateUpdated` and it works fine getting and setting values of other form fields. But how can I trigger the rerendering of a ViewField without any direct connection to the form itself?

**Comment by danharrin** [comment]
If the TextInput is `reactive()` or `lazy()` then it will rerender the form each time it changes

**Comment by OvalMedia** [comment]
Are you referring to the other fields that should trigger the update (quantities, prices, positions...)?
They already ARE `reactive()`.
I should add that the total field is part of the `invoice` form while the reactive input fields are part of a `repeater` derived from the related invoice positions.

**Comment by OvalMedia** [comment]
Ok, I think I am closing in on my mistake. I can confirm the form IS refreshing properly but the content of the ViewField is not. Inside the livewire component I am using `$this->record` to access the underlying invoice model. But that does not seem to reflect the current values in the form ("state"?) but rather the last state of the database. 

How can I access a model (invoice) that represents the current forms data that has not been persisted to the database yet?

**Reply by danharrin** [reply]
As I said in the other issue `$get()` is used to fetch unsaved data from other form components. You can access `$get()` using `$evaluate(fn ($get) => ...)` in the view

**Reply by OvalMedia** [reply]
I understand but $get only seems to return the value of single fields. Since I am invoking a function (`renderTotals`) that loops over all children of the invoice (positions) I thought I somehow could use the actual model with current (unpersisted) data? Isn't that the idea behind data binding?

**Reply by danharrin** [reply]
Data isnt stored directly on the model, because the format often differs to what is eventually saved in the database. But it's quite difficult to help you without code examples. Ideally I'd like a minimal repository where I can see what you want to achieve with all the code you're trying to use

**Reply by OvalMedia** [reply]
Thank you, i appreciate your effort. But seting up a whole new project to make the point is a bit overkill and I can't publish the project I currently have, there is too much sensitive stuff in it already. I will have to try something else.

**Reply by danharrin** [reply]
Okay, but I'm not going to be able to help you without seeing actual code. You don't need to publish the project, just reproduce the thing you want to do

**Reply by OvalMedia** [reply]
I am trying a different approach for now.

**Comment by markosole** [comment]
Hi @danharrin,

I am sort of stuck on the silimar block. Using ViewField to show the view with Radio button selection of items. I am trying to create custom component that will allow me to select a image as a Radio. Anyway... 

Field is:

```php
ViewField::make('selectedCity')
                    ->view('forms.components.select-city')
                    ->afterStateUpdated(function (?string $state, ?string $old) {
                        ray("Value changed to: $state");
                    })
                    ->live()
                    ->viewData([
                        'min' => 1,
                        'max' => 5,
                        'cities' => ['Zagreb', 'Dublin', 'London', 'Sarajevo', 'Cape Town'],
                    ]),
```

And View is:

```php

<x-dynamic-component
    :component="$getFieldWrapperView()"
    :field="$field"
>
    <div x-data="{ selectedCity: null }">
        <!-- Interact with the `selectedCity` property in Alpine.js -->
        Select option listed below:

        <!-- List of cities passed from the backend -->
        @foreach ($cities as $city)
            <label>
                <input type="radio" name="city" value="{{ $city }}" x-model="selectedCity">
                {{ $city }} <!-- Display the city name -->
            </label>
        @endforeach

        <!-- Output the selected city (for testing) -->
        <p>Selected city: <span x-text="selectedCity"></span></p>
    </div>
</x-dynamic-component>

```

What I would like to do is maybe run a "afterStateUpdated" and update the state of other field(s) or run cistom function that will do something else in Form. What would be the way to do it?

**Reply by leandrocfe** [reply]
There is a plugin that you can use radio buttons with images
https://filamentphp.com/plugins/ibrahim-bougaoua-radio-btn-image

**Reply by markosole** [reply]
True, I have looked into it and I could not install it due dependecies issues. I did look into replicating it but no joy with that :/
